crates/autopilot/src/domain/settlement/trade/mod.rs

```diff
@@ -159,12 +159,20 @@ pub struct ExecutedFee {
     pub network: eth::SellTokenAmount,
     /// Breakdown of protocol fees. Executed protocol fees are in the same order
     /// as policies are defined for an order.
-    pub protocol: Vec<(eth::SellTokenAmount, fee::Policy)>,
+    pub protocol: Vec<(ExecutedProtocolFee, fee::Policy)>,
+}
+
+#[derive(Debug, Clone)]
+pub struct ExecutedProtocolFee {
+    /// Token in which the fee was paid
+    pub token: eth::TokenAddress,
+    /// Amount of the fee
+    pub amount: eth::SellTokenAmount,
```

Comment:
Is this the amount in the token in which the fee was paid or in the sell token? If the former, the type is absolutely misleading. If the latter, token should probably just be called sellToken (if it was me, I'd still prefer to make SellTokenAmount -> TokenAmount)

Also, can we add the policy here. Seems like it's often used together, so it should probably be a part of the struct.

Reply:
refactored, please check if i adressed everything